### PR TITLE
Issues #402, #438, #522

### PR DESCRIPTION
**Pull request by ognjenns**

### CODE REVIEW

File: Examples/restcomm-olympus-nopod/restcomm-olympus/Utils.m [hunk 1]
```diff
@@ -34,20 +51,20 @@ + (void) setupUserDefaults
...
-                                    @"turn-password" : @"4e89a09e-bf6f-11e5-a15c-69ffdcc2b8a7",  // @"4080218913"
-                                    @"signaling-secure" : @(YES),  // by default signaling is secure
-                                    @"signaling-certificate-dir" : @"",
+                                    kIsFirstTimeKey : @(YES),
+                                    kPendingInterappKey : @"",  // has another app sent us a URL to call?
+                                    kSipIndentificationKey : @"",  //@"sip:ios-sdk@cloud.restcomm.com",
+                                    kSipPasswordKey : @"",
+                                    kSipRegistrarKey : @"cloud.restcomm.com",
+                                    kTurnEnabledKey : @(YES),
+                                    kTurnUrlKey : @"https://service.xirsys.com/ice",  // @"https://computeengineondemand.appspot.com/turn",
```
Comment (atsakiridis) [inline Examples/restcomm-olympus-nopod/restcomm-olympus/Utils.m]: Lets use a xirsys V3 url as discussed in my other comment for kTurnUrlKey. I believe it should be:

`https://es.xirsys.com/_turn/`

Just make sure that you test it ;)

File: Examples/restcomm-olympus-nopod/restcomm-olympus/Utils.m [hunk 2]
```diff
@@ -25,6 +25,23 @@
 #import "RCUtilities.h"
 
 
+//keys
+NSString *const kContactKey = @"contacts";
```
Comment (atsakiridis) [inline Examples/restcomm-olympus-nopod/restcomm-olympus/Utils.m]: Good idea on introducing constants for these keys 👍 . Actually, at some point we need to introduce them within the SDK to make it easier to use keys for the parameters argument. But let's leave that for later

File: RestCommClient/Classes/RCDevice.h [hunk 3]
```diff
@@ -116,9 +125,22 @@ extern NSString* const RCDeviceCapabilityClientNameKey;
...
+ *    <b>ice-servers</b> NSArray of NSDictionary data of the STUN and TURN servers; This key is mandatory if ice-config-type is 'kCustom'; Example with 2 servers:
+            parameter to pass (Array):
+            @[
+                @{ @"url":@"URL", @"username":@"USERNAME_HERE", @"credential":@"KEY_HERE"},
+                @{ @"url":@"URL" }
+            ]
+ 
  *  @param delegate        Delegate of RCDevice
- *
+ *  @param iceConfigType   Type of the ICE configuration
```
[the remark this answers is not in the document]
Comment (ognjenns) [reply RestCommClient/Classes/RCDevice.h]: Yeap, forgot to remove it.

File: RestCommClient/Classes/SipManager/SipManager.mm [hunk 4]
```diff
@@ -562,7 +561,11 @@ - (void)handleCall
     if (!self.media) {
         [self initializeAudioSession];
         
-        self.media = [[MediaWebRTC alloc] initWithDelegate:self parameters:self.params];
+        
+        //NOTE: We dont want to create separate header for iceConfigType
+        //so, we created a ICEConfigTypeForMedia and we should map ICEConfigType to it//
```
[the remark this answers is not in the document]
Comment (ognjenns) [reply RestCommClient/Classes/SipManager/SipManager.mm]: Nope. WIll remove it.

**Pull request merged**